easypost/beta/referral.py

```diff
@@ -0,0 +1,82 @@
+from typing import Optional
+
+import requests
+
+from easypost.easypost_object import convert_to_easypost_object
+from easypost.requestor import (
+    RequestMethod,
+    Requestor,
+)
+
+
+class Referral:
+    @classmethod
+    def add_credit_card(
+        cls,
+        api_key: Optional[str] = None,
+        primary_or_secondary="primary",
+        **params,
+    ) -> dict:
+        """Add credit card to a referral user."""
+        easypost_key = cls.__retrieve_ep_key(api_key=api_key)
+        stripe_token = cls.__stripe_create_token(easypost_key=easypost_key["public_key"], **params).json()
+        response = cls.__ep_create_card(
+            api_key=api_key,
+            stripe_object_id=stripe_token.get("id"),
+            priority=primary_or_secondary,
+        )
+        return convert_to_easypost_object(response)
+
+    @classmethod
+    def __retrieve_ep_key(
+        cls,
+        api_key: Optional[str] = None,
+    ) -> dict:
+        """Retrieve EasyPost's Stripe Public API Key."""
+        requestor = Requestor(local_api_key=api_key)
+        public_key, _ = requestor.request(method=RequestMethod.GET, url="/partners/stripe_public_key", beta=True)
+        return public_key
+
+    @classmethod
+    def __stripe_create_token(
+        cls,
+        easypost_key=None,
+        **params,
+    ) -> requests.Response:
+        """Get Token From Stripe."""
+        headers = {
+            "Authorization": f"Bearer {easypost_key}",
+            "Content-type": "application/x-www-form-urlencoded",
+        }
+        credit_card_dict = {
+            "card": {
+                "number": params.get("number", ""),
+                "exp_month": params.get("exp_month", ""),
+                "exp_year": params.get("exp_year", ""),
+                "cvc": params.get("cvc", ""),
+            }
+        }
+
+        encoded_form = Requestor.form_encode_params(credit_card_dict)
+
+        url = "https://api.stripe.com/v1/tokens"
+        stripe_response = requests.post(url, params=encoded_form, headers=headers)
+        return stripe_response
+
+    @classmethod
+    def __ep_create_card(
+        cls,
+        api_key: Optional[str] = None,
+        stripe_object_id=None,
+        priority="primary",
+    ) -> dict:
+        """Submit Stripe Token to EasyPost."""
```

Comment:
There are a bunch of "Stripe tokens", please specify that this is to submit the `Stripe credit card token`